pkg/services/contexthandler/contexthandler.go

```diff
@@ -445,29 +448,31 @@ func (h *ContextHandler) initContextWithToken(reqContext *models.ReqContext, org
 		getTime = time.Now
 	}
 
-	// Check whether the logged in User has a token (whether the User used an OAuth provider to login)
-	oauthToken, exists, _ := h.oauthTokenService.HasOAuthEntry(ctx, queryResult)
-	if exists {
-		// Skip where the OAuthExpiry is default/zero/unset
-		if !oauthToken.OAuthExpiry.IsZero() && oauthToken.OAuthExpiry.Round(0).Add(-oauthtoken.ExpiryDelta).Before(getTime()) {
-			reqContext.Logger.Info("access token expired", "userId", query.UserID, "expiry", fmt.Sprintf("%v", oauthToken.OAuthExpiry))
-
-			// If the User doesn't have a refresh_token or refreshing the token was unsuccessful then log out the User and Invalidate the OAuth tokens
-			if err = h.oauthTokenService.TryTokenRefresh(ctx, oauthToken); err != nil {
-				if !errors.Is(err, oauthtoken.ErrNoRefreshTokenFound) {
-					reqContext.Logger.Error("could not fetch a new access token", "userId", oauthToken.UserId, "error", err)
-				}
-
-				reqContext.Resp.Before(h.deleteInvalidCookieEndOfRequestFunc(reqContext))
-				if err = h.oauthTokenService.InvalidateOAuthTokens(ctx, oauthToken); err != nil {
-					reqContext.Logger.Error("could not invalidate OAuth tokens", "userId", oauthToken.UserId, "error", err)
-				}
-
-				err = h.AuthTokenService.RevokeToken(ctx, token, false)
-				if err != nil && !errors.Is(err, models.ErrUserTokenNotFound) {
-					reqContext.Logger.Error("failed to revoke auth token", "error", err)
+	if h.features.IsEnabled(featuremgmt.FlagAccessTokenExpirationCheck) {
+		// Check whether the logged in User has a token (whether the User used an OAuth provider to login)
+		oauthToken, exists, _ := h.oauthTokenService.HasOAuthEntry(ctx, queryResult)
+		if exists {
```

Comment:
refactor for later
- could we make this check for `!exists` to make the code more to the left 👯 https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88